Неделя #4 Работаем с сагами. Реализуем заказ такси.

cosmieel opened the pull request:
Подключил redux-saga. Перенёс сетевые запросы в саги.
Реализовал авторизацию, отправку платёжных данных на сервер, получение списка доступных адресов, получение точек маршрута с сервера.
Реализовал функционал заказа такси и построения маршрута.
Покрыл саги и редьюсеры тестами.

![Снимок2](https://user-images.githubusercontent.com/32985372/95010035-06aa8480-062f-11eb-8f39-1e2918df561f.PNG)
![Снимок](https://user-images.githubusercontent.com/32985372/95010036-09a57500-062f-11eb-89bf-23a87786224e.PNG)

cosmieel committed:
feature(components): подключение redux-saga для компонентов аутентификации

cosmieel committed:
feature(components): подключение redux-saga к странице профиля

cosmieel committed:
feature(product): добавление функционала вызова такси

cosmieel committed:
test: написание тестов для саг и обновление тестов редьюсеров

cosmieel committed:
test(components): обновление тестов компонентов

Обновление тестов старых компонентов + написание тестов для компонентов формы заказа такси

Danil00524 commented on src/modules/address/middleware.js:
```diff
@@ -0,0 +1,5 @@
+export const fetchAddressMiddleware = async () => {
+    const response = await fetch(`https://loft-taxi.glitch.me/addressList`);
```
https://loft-taxi.glitch.me/ - должно быть в middleware. Основная задумка здесь в том, что если поменяется хост, поменять его на сайте в одном месте, а не в 10-ти.

Danil00524 commented on src/modules/auth/middleware.js:
```diff
@@ -1,72 +1,13 @@
-    fetchSignUpFailure
-} from './actions'
+export const authRequestMiddleware = async (payload, path) => {
+    const response = await fetch(`https://loft-taxi.glitch.me/${path}`, {
```
Тут тоже самое. Просто тебе нужна ф-ция обертка, которая будет вызываться каждый раз, а туда передавать нужные параметры.

cosmieel committed:
fix: фикс

Обновление тестов старых компонентов + написание тестов для компонентов формы заказа такси

cosmieel committed:
refactor(product): перенос запросов к серверу в отдельный файл
